src/combineAll.js

```diff
@@ -0,0 +1,8 @@
+const compose  = require('ramda/src/compose')
+const identity = require('ramda/src/identity')
+const juxt     = require('ramda/src/juxt')
+const mergeAll = require('ramda/src/mergeAll')
+
+// combineAll : [a... -> { k: v }] -> { k: v } -> { k: v }
```

Comment:
should this be something like:
```haskell
combineAll :: [ (Object -> Object) ] -> Object -> Object
```
If I get this straight you provide it a list of unary endomorphisms in Object, pass it an Object and get back an `Object`. (List of functions that take one arg, as you can only apply one an map an Object to an Object)

Is that the intent?

Reply:
I guess it should be:
```haskell
combineAll :: [ ((Object, ...) -> Object) ] -> (Object, ...) -> Object
```

Reply:
Because juxt is polyadic (n-ary) and the functions in the list are also polyadic

Reply:
😕 But the second argument isn't a list.

```haskell
combineAll :: [ ((Object, ...) -> Object) ] -> Object -> Object
```

```javascript
combineAll([ always({ foo: 1 }), always({ bar: 2 }) ], { baz: 3 })
//=> { foo: 1, bar: 2, baz: 3 }
```

❓

Reply:
```javascript
const f =
  ({ a }, { c }) => ({ ac: a + c })

const g =
  ({ b }, { d }) => ({ bd: b + d })

combineAll([ f, g ])({ a: 10, b: 15 }, { c: 0, d: 100 })
```

Reply:
Ooh. Got it. Thanks!

Reply:
Now if you want to go full unary, I do not know if we have a `thrush` combinator but you could:

```javascript
const compose  = require('ramda/src/compose')
const identity = require('ramda/src/identity')
const flip = require('ramda/src/flip')
const map = require('ramda/src/map')
const mergeAll = require('ramda/src/mergeAll')

// thrush :: a -> (a -> b) -> b
const thrush =
  x => fn => fn(x)

// combineAll :: [ (Object -> Object) ] -> Object -> Object
const combineAll = fns => compose(
  mergeAll,
  flip(map)([ identity, ...fns ]),
  thrush
)

const f =
  ({ a, b }) => ({ ab: a + b })

const g =
  ({ b, c }) => ({ bc: b + c })

combineAll([ f, g ])({ a: 1, b: 10, c: 21 })
// { a: 1, b: 10, c: 21, ab: 11, bc: 31 }
```

Reply:
Also if you are looking for a typsafe approach, you can do something like this with crocks like:

```js
const Maybe = require('crocks/Maybe')

const chain = require('crocks/pointfree/chain')
const compose  = require('crocks/helpers/compose')
const flip = require('crocks/combinators/flip')
const identity = require('crocks/helpers/identity')
const isObject = require('crocks/predicates/isObject')
const map = require('crocks/pointfree/map')
const option = require('crocks/pointfree/option')
const safe = require('crocks/Maybe/safe')
const traverse = require('crocks/pointfree/traverse')

const mergeAll = require('ramda/src/mergeAll')

const f =
  ({ a, b }) => ({ ab: a + b })

const g =
  ({ b, c }) => ({ bc: b + c })

// thrush :: a -> (a -> b) -> b
const thrush =
  x => fn => fn(x)

// combineAll :: [ (a -> b) ] -> a -> Maybe Object
const combineAll = fns => compose(
  map(mergeAll),
  traverse(Maybe.of, safe(isObject)),
  flip(map)([ identity, ...fns ]),
  thrush
)

// flow :: [ (a -> b) ] -> a -> Maybe Object
const flow = fn => compose(
  option({}),
  chain(combineAll(fn)),
  safe(isObject)
)

flow([ f, g ])({ a: 1, b: 10, c: 21 })
// { a: 1, b: 10, c: 21, ab: 11, bc: 31 }
```

If either the input or all the results of all the function call are not all Objects, if returns a default empty Object.

It is extra work make gives us guarantees that the other flow does not. But this will be free of exceptions (unless you do not pass a list of functions, but that is just chaining another traversal, if you wanna go that deep, you can even fuse traversals to keep it one pass.

Not a suggestion, just something to consider for fun and safety.

Reply:
But this is good the way that it is IMO, mergeAll can handle an array of whateves. we are good there. it just shakes out.

Reply:
![](https://morbotron.com/video/S03E10/bwjMV7ESxL4ujc4Q3mThnz7YVIQ=.gif)